File: mobile/androidApp/src/main/kotlin/com/mentora/android/ui/components/QuestionCard.kt
package com.mentora.android.ui.components

import androidx.compose.foundation.BorderStroke
import androidx.compose.foundation.layout.Arrangement
import androidx.compose.foundation.layout.Column
import androidx.compose.foundation.layout.padding
import androidx.compose.material3.MaterialTheme
import androidx.compose.material3.Surface
import androidx.compose.material3.Text
import androidx.compose.runtime.Composable
import androidx.compose.ui.Modifier
import androidx.compose.ui.res.stringResource
import com.mentora.android.R
import com.mentora.android.theme.MentoraDimens

/**
 * `design-system/COMPONENTS.md` § QuestionCard (line 716-720): "same shell as a StatCard" (radius
 * `radius.large`, border `color.border.default`, padding `space.5`), containing a "Question N of M"
 * progress indicator (`typography.caption`) paired with a slim [MentoraProgressBar], then the
 * question text (`heading.h4`).
 *
 * Pure presentational — [questionNumber]/[totalQuestions]/[questionText] are plain values; Task 14
 * wires real quiz domain models into these parameters, not this component's job.
 */
@Composable
fun QuestionCard(
    questionNumber: Int,
    totalQuestions: Int,
    questionText: String,
    modifier: Modifier = Modifier,
    // T19 review fix (LOW-1, D94): generalizing ErrorState.retryLabel's own HIGH fix — see that
    // component's kdoc. `.toString()` (not raw `%d`) keeps numerals Western per LOCALIZATION.md § 8.
    progressLabel: String = stringResource(
        R.string.question_card_progress_label,
        questionNumber.toString(),
        totalQuestions.toString(),
    ),
) {
    Surface(
        modifier = modifier,
        shape = MaterialTheme.shapes.medium, // radius.large (16).
        color = MaterialTheme.colorScheme.surface,
        border = BorderStroke(MentoraDimens.borderWidthDefault, MaterialTheme.colorScheme.outlineVariant),
    ) {
        Column(
            modifier = Modifier.padding(MentoraDimens.spacing.space5),
            verticalArrangement = Arrangement.spacedBy(MentoraDimens.spacing.space2),
        ) {
            Text(
                text = progressLabel,
                style = MaterialTheme.typography.labelSmall, // typography.caption.
                color = MaterialTheme.colorScheme.onSurfaceVariant,
            )
            MentoraProgressBar(
                progress = if (totalQuestions > 0) questionNumber.toFloat() / totalQuestions.toFloat() else 0f,
                contentDescriptionLabel = progressLabel,
            )
            Text(
                text = questionText,
                style = MaterialTheme.typography.titleLarge, // heading.h4.
                color = MaterialTheme.colorScheme.onSurface,
            )
        }
    }
}

File: mobile/iosApp/iosApp/Components/QuestionCard.swift
import SwiftUI
import shared

// Phase 5 Task T11 slice 3 (Component Kit B) -- `design-system/COMPONENTS.md § QuestionCard` (lines
// 716-720): "same shell as a StatCard" (radius `radius.large`, border `color.border.default`, padding
// `space.5`), containing a "Question N of M" progress indicator (`typography.caption`) paired with a
// slim `MentoraProgressBar`, then the question text (`heading.h4`). Ported directly from Android's own
// `QuestionCard.kt`.
//
// Pure presentational -- `questionNumber`/`totalQuestions`/`questionText` are plain values; a later
// task wires real quiz domain models into these parameters, not this component's job.
enum QuestionCardRules {
    /// `totalQuestions > 0 ? questionNumber / totalQuestions : 0` -- a directly-testable divide-by-zero
    /// guard, mirroring Android's own identical inline conditional exactly.
    static func progressFraction(questionNumber: Int, totalQuestions: Int) -> Double {
        guard totalQuestions > 0 else { return 0 }
        return Double(questionNumber) / Double(totalQuestions)
    }
}

struct QuestionCard: View {
    let questionNumber: Int
    let totalQuestions: Int
    let questionText: String
    /// DISCLOSED DEVIATION from Android's own default-parameter convenience -- see
    /// `CourseProgressCard.swift`'s identical `resumeLabel` doc comment for the full rationale. Real,
    /// pre-existing `question_card_progress_label` key (`%1$@`/`%2$@`, confirmed present in
    /// `Resources/Localizable.xcstrings` before this task began), REQUIRED here, always resolved and
    /// passed in by the caller via `MentoraStrings.text(_:locale:_:)`.
    var progressLabel: String

    var body: some View {
        VStack(alignment: .leading, spacing: MentoraSpacing.space2) {
            Text(progressLabel)
                .mentoraFont(.caption)
                .foregroundStyle(Color.mentoraTextSecondary)
            MentoraProgressBar(
                progress: QuestionCardRules.progressFraction(questionNumber: questionNumber, totalQuestions: totalQuestions),
                accessibilityLabel: progressLabel
            )
            Text(questionText)
                .mentoraFont(.h4)
                .foregroundStyle(Color.mentoraTextPrimary)
        }
        .padding(MentoraSpacing.space5)
        .background(Color.mentoraSurfaceDefault)
        .clipShape(MentoraShape.large) // radius.large (16).
        .overlay {
            MentoraShape.large.strokeBorder(Color.mentoraBorderDefault, lineWidth: MentoraBorderWidth.default)
        }
    }
}
